samples/documentation/lulesh_sample1_unix_commented.yaml

```diff
@@ -0,0 +1,234 @@
+    # enforcing fixed formatting for output files, or fixed formatting for
+    # components of steps which follow fixed formats.
+    #-------------------------------
+    labels:
+        outfile: $(SIZE.label).$(ITERATIONS.label).log
+
+    #-------------------------------
+    # Dependencies
+    #-------------------------------
+    # Dependencies represent external artifacts that should be present before a
+    # workflow can run. These are things like acquirable inputs from a directory
+    # or a repository such as input files for programs, code, data, etc..
+
+    # NOTE: Currently there are only two types of dependencies:
+    # path - verifies the existence of the specified path before execution.
+    # git  - clones the specified repository before execution of the study.
+    #-------------------------------
+    dependencies:
+      git:
+        - name: LULESH
+          path: $(OUTPUT_PATH)
+          url: https://github.com/LLNL/LULESH.git
+
+####################################
+# Study Block (Required)
+####################################
+# The study block is where the steps in the workflow are defined. This section
+# of the specfication represents the unexpanded set of tasks that the study
+# is composed of.
+####################################
+study:
+    #-------------------------------
+    # Study Steps
+    #-------------------------------
+    # Represents a task to be completed in the study's workflow.
+    #-------------------------------
+      # 'make-lulesh' is an example of a singular step because the task
+      # does not contain parameters (more on those later).
+      # name: Unique name that identifies the task. (Required)
+    - name: make-lulesh
+      # description: A general description of what this step is intended to do.
+      # (Required)
+      description: Build the serial version of LULESH.
+      # run: Properties that describe the actual specification of the task.
+      run:
+          # cmd: The actual task to be executed. (Required)
+          cmd: |
+            cd $(LULESH)
+            sed -i 's/^CXX = $(MPICXX)/CXX = $(SERCXX)/' ./Makefile
+            sed -i 's/^CXXFLAGS = -g -O3 -fopenmp/#CXXFLAGS = -g -O3 -fopenmp/' ./Makefile
+            sed -i 's/^#LDFLAGS = -g -O3/LDFLAGS = -g -O3/' ./Makefile
+            sed -i 's/^LDFLAGS = -g -O3 -fopenmp/#LDFLAGS = -g -O3 -fopenmp/' ./Makefile
+            sed -i 's/^#CXXFLAGS = -g -O3 -I/CXXFLAGS = -g -O3 -I/' ./Makefile
+            make clean
+            make
+          # depends: A list of task names that this step is dependent on.
+          # The 'make-lulesh' step is not dependent on other tasks, and so will
+          # be executed at the start of the study.
+          depends: []
+          # NOTE: There are additional step entries for setting up a step for
+          # remote execution. For examples see 'launcher_tokens.yaml' in the
+          # 'samples/documentation' subdirectory in the maestrowf repo.
+

```

Comment:
I presume you're going to eventually have a more complete, formal list of all of the options, right?

Reply:
Yeah, that the eventual goal. The comments here are meant to help users make sense of this particular spec and to get the generals in place.